### ISSUE TITLE
(core): Python package aws-cdk-lib is missing the __version__ attribute

### ISSUE THREAD

**Issue by mwebber** [aws/aws-cdk#28875]
### Describe the bug

The normal convention in the Python world is for pip-installed packages to report their version number via the `__version__` attribute. However, `aws_cdk` does not currently support this, so code cannot introspect its environment to check whether, for example, the expected version of `aws_cdk` is being used.

### Expected Behavior

I expected it to work in the same was as most other Python packages, e.g
```
# this works
>>> import boto3
>>> boto3.__version__
'1.34.3'
```

Thus, I should be able to do
```
# this doesn't work, but should
>>> import aws_cdk
>>> aws_cdk.__version__
'2.113.0'
```

### Current Behavior

```
# current behavior
>>> import aws_cdk
>>> aws_cdk.__version__
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'aws_cdk' has no attribute '__version__'
```

### Reproduction Steps

This can be tested in a stand-alone environment easily enough

Show that __version__ not defined
```
python3
import aws_cdk
print(aws_cdk.__version__)
# AttributeError thrown here
exit()
```

The version _is_ available in the package metadata, it's just not exposed as a python attribute
```
pip show aws-cdk-lib
Name: aws-cdk-lib
Version: 2.113.0
Summary: Version 2 of the AWS Cloud Development Kit library
Home-page: https://github.com/aws/aws-cdk
Author: Amazon Web Services
Author-email:
License: Apache-2.0
Location: /xxx/.venv/lib/python3.11/site-packages
Requires: aws-cdk.asset-awscli-v1, aws-cdk.asset-kubectl-v20, aws-cdk.asset-node-proxy-agent-v6, constructs, jsii, publication, typeguard
```

### Additional Information/Context

The following workaround can be used to get what appears to be the CDK version, but it's clearly an irregular approach.
```
from aws_cdk._jsii import __jsii_assembly__ as jsii_assembly
print("*** have", jsii_assembly.version)
```

### CDK CLI Version

2.113.0 (build ccd534a)

### Node.js Version

v20.10.0

### OS

macOS 13.4.1

### Language

Python

### Language Version

Python (3.11.6)

**Comment by pahud** [comment]
Makes sense. Thank you for the feature request.

**Comment by mwebber** [comment]
@pahud After further investigation, I found https://github.com/pallets/flask/issues/5230, which suggests that `__version__`  is superseded by `importlib.metadata.`.

This already works:
```
from importlib import metadata
[x.version for x in metadata.distributions() if x.name=='aws-cdk-lib'][0]
'2.113.0'
```
so maybe this ticket can be closed as unnecessary.
